### DISCUSSION TITLE
Keeping intermediate files during compile

### DISCUSSION THREAD

**Discussion by sammo3182** [quarto-dev/quarto-cli#1730] [Q&A] [answered]
I try to keep the intermediate files, such as images, during pdf compiling. It seems that `latex-clean: false` helps. But no matter where I put it in the YAML, the intermediate folder is cleaned after the rendering. Here's an example. Is there a way to keep the `test_files` folder from deleting?

```
---
title: "test"
format: 
  pdf: 
    latex-clean: false
---

A test 
```

**Comment by mcanouil** [comment, marked as answer]
You are looking for `keep-tex`, there is also `keep-ipynb` and `keep-md` depending on the format.

See https://quarto.org/docs/reference/formats/pdf.html#rendering

**Reply by sammo3182** [reply]
Aha, work! so when should the `latex-clean: false` be used?

**Reply by mcanouil** [reply]
Possibly never by end-user (could be useful when developing an extension format), since in my understanding the `latex-clean` is more about the latex dependencies required to compile a document (e.g., packages, bibliography intermediate files, etc.).
